### ISSUE TITLE
Move, delete and connect nodes which are part of a way

### ISSUE THREAD

**Issue by cyberchrime** [Helium314/StreetComplete#593]
**Use case**

Today we discovered a way which should have been connected to another street, but just ended in the middle of now where. If I could connect both streets, it would allow navigation apps to properly create routes.

**Proposed Solution**

If it would be possible to select the separate nodes of a way, move them and possibly connect them to other streets, it would be very helpful. As the possibility to create nodes as part of a way was added in issue #414, I assume that some basic functionality to edit the nodes of ways is already available.

I think a dedicated overlay could be useful for this feature. Once the user switches to this overlay, the nodes of the ways are displayed and can be selected. Once selected, the node can be deleted or moved. If moved onto another way, the node connects both ways.

**Comment by rusty-snake** [comment]
Moving notes is already possible with custom overlays (see "All nodes" https://github.com/Helium314/SCEE/discussions/574) and the usual 29,9m steps (I should open an issue for SC**EE** to raise this limit, I often have to move POIs 30-50m because the physical object was moved from oneside of a parking place to the other or because of an estimated position at creation time).

Deleting nodes (that a part of a way) from custom overlays does not seem to work.

Connecting does very likely not work (I did not tested it).

> possibility to create nodes as part of a way was added in issue https://github.com/Helium314/SCEE/issues/414

OT: Which sometimes (have to inspect it better) seems to create self-intersections.

> Move, delete and connect nodes which are part of a way
> I think a dedicated overlay could be useful for this feature.

The benefit of a dedicated overlay is a better commit message.

**Comment by HolgerJeromin** [comment]
> I should open an issue for SCEE to raise this limit

Already there #402

**Comment by HolgerJeromin** [comment]
For such complex operations like node deletion / merging you need https://vespucci.io/

**Comment by rusty-snake** [comment]
> https://vespucci.io/

Unfortunately vespucci cannot be used offline because you cannot download larger areas (10x10km) in advance.

**Comment by Helium314** [comment]
If someone wants to make the necessary changes and deal with follow-up issues, you're welcome. Currently it's certainly too much for me.
Best wait until SC migration to compose is done (to avoid re-doing the UI), possibly also MapLibre migration.

> Unfortunately vespucci cannot be used offline because you cannot download larger areas (10x10km) in advance.

I think downloading 100 km² is not a requirement for using an editor offline. Even when travelling a lot I don't think I ever had such large area at once downloaded in SCEE.

**Comment by mnalis** [comment]
> Unfortunately vespucci cannot be used offline because you cannot download larger areas (10x10km) in advance.

@rusty-snake Actually, you can, just not as easy. See Vespucci [offline guide](https://vespucci.io/tutorials/offline_guide/) for details. I often download whole of Croatia beforehand when I expect to need to work offline for extended period of time.

> [...] and the usual 29,9m steps (I should open an issue for SCEE to raise this limit

I think there is already https://github.com/Helium314/SCEE/issues/402 

---

As for the original request, I personally think that _"possible to select the separate nodes of a way, move them and possibly connect them to other streets"_ is probably **too much** for SCEE... 
Because, if you can **connect** them, you surely should be able to **disconnect** them too, right? And connecting to another way implies ability to extend ways, so you should be able to split ways too. And why not being able to create new ways and delete old ones too then when you've gone that far?
 
Also when you connect one way to other way being in relation(s), do you extend relation to that road extension? And if you do, which one? All of them? You should certainly give user a choice... And if you do all that, you'd have re-implemented most of Vespucci. :smiley_cat:

**Comment by Helium314** [comment]
@mnalis yes, such things are roughly what I thought of when I wrote "_and deal with follow-up issues_".
